Docs/source/Software/RTOS.rst

```diff
@@ -4,63 +4,49 @@ BPS RTOS Layer
+    or take control of a resource in the program, ensuring that multiple threads don't write to a value at the same time.
 
-Mutex Pend - ``RTOS_BPS_MutexPend(BPS_OS_MUTEX* mutex, BPS_OS_TICK timeout, BPS_OS_OPT opt)``
-    ``RTOS_BPS_MutexPend()`` pends a mutex, and requires three parameters:
-
-    1) A pointer to a BPS_Mutex object, ``BPS_OS_Mutex* mutex``.
-    2) A time to wait on the pend, ``BPS_OS_TICK timeout``.
-    3) A pend option, ``BPS_OS_OPT opt``.
-
-    ``RTOS_BPS_MutexPend()`` returns nothing.
+Mutex Pend - ``RTOS_BPS_MutexPend(BPS_OS_MUTEX* mutex, BPS_OS_OPT opt)``
+    The ``RTOS_BPS_MutexPend`` function pends a specified mutex object. Pending a mutex
+    takes control of the mutex, and allows you to write to the resource as needed. When done with the 
+    resource, a call to ``MutexPost`` should be made, informing the RTOS that you are releasing the
+    lock on the mutex, marking it as available again.
 
 Mutex Post - ``RTOS_BPS_MutexPost(BPS_OS_MUTEX* mutex, BPS_OS_OPT options)``
-    ``RTOS_BPS_MutexPost()`` posts a mutex, and requires two parameters:
-
-    1) A pointer to a BPS_Mutex object, ``BPS_OS_Mutex* mutex``.
-    2) A pend option, ``BPS_OS_OPT opt``.
-
-    ``RTOS_BPS_MutexPost()`` returns nothing.
-
-Task Create - ``RTOS_BPS_TaskCreate(BPS_OS_TCB* p_tcb, char* p_name, void* p_task, void* p_arg, uint8_t prio, BPS_CPU_STK* p_stk_base, uint64_t stk_size)``
-    ``RTOS_BPS_TaskCreate()`` creates an RTOS task, and has several parameters. This function will be refactored upon moving to FreeRTOS.
+    The ``RTOS_BPS_MutexPost`` function posts a specified mutex object. Posting a mutex releases
+    the mutex lock on the resource, marking it as available to the RTOS. Once you relinquish this mutex,
+    you SHOULD NOT write to its associated resource until you successfully ``Pend`` the resource again.
 
-    1) A pointer to a TCB for the task, ``BPS_OS_TCB* p_tcb``.
-    2) A pointer to the name of the task, ``char* p_name``.
-    3) A pointer to the function to call when the task executes, ``void* p_task``.
-    4) A pointer to the arguments of the task function, ``void* p_arg``.
-    5) Priority level of the task, ``uint8_t prio``.
-    6) A pointer to the stack base, ``BPS_CPU_STK* p_stk_base``.
-    7) The stack size, ``uint64_t stk_size``.
+Task Create - ``RTOS_BPS_TaskCreate(BPS_OS_TCB *p_tcb, char *p_name, void *p_task, void *p_arg, uint8_t prio, BPS_CPU_STK *p_stk_base, uint64_t stk_size)``
+    The ``RTOS_BPS_TaskCreate`` function, as the name implies, creates a Task for the RTOS. 
+    `**RTOS Tasks** <https://www.digikey.com/en/maker/projects/introduction-to-rtos-solution-to-part-3-task-scheduling/8fbb9e0b0eed4279a2dd698f02ce125f>`_  are a powerful feature of using the RTOS.
+    Tasks are executed in sequential order depending on their priorities; if tasks have the same priority, then 
+    they are executed in a round-robin fashion. Tasks allow us to run specific code at specific times or intervals
```

Comment:
Maybe put more detail about scheduling for each task (ie the time slice for each task)? But that may be too detailed for the sake of these docs.

Reply:
I don't think it needs to be included. If someone was using this code, it should be assumed they know how an RTOS basically works.